wrappers/core18.go

```diff
@@ -231,49 +250,54 @@ func AddSnapdSnapServices(s *snap.Info, inter interacter) error {
 		// systemd version, where older versions (eg 229 in 16.04) would
 		// error out unless --force is passed, while new ones remove the
 		// symlink and create a new one.
-		enabled, err := sysd.IsEnabled(unit)
-		if err != nil {
-			return err
-		}
-		if enabled {
-			continue
+		if !opts.Preseeding {
+			enabled, err := sysd.IsEnabled(unit)
+			if err != nil {
+				return err
+			}
+			if enabled {
+				continue
+			}
 		}
 		if err := sysd.Enable([]string{unit}); err != nil {
 			return err
 		}
 	}
 
-	for _, unit := range changed {
-		// Some units (like the snapd.system-shutdown.service) cannot
-		// be started. Others like "snapd.seeded.service" are started
-		// as dependencies of snapd.service.
-		if snapdSkipStart(snapdUnits[unit].(*osutil.MemoryFileState).Content) {
-			continue
-		}
-		// Ensure to only restart if the unit was previously
-		// active. This ensures we DTRT on firstboot and do
-		// not stop e.g. snapd.socket because doing that
-		// would mean that the snapd.seeded.service is also
-		// stopped (independently of snapd.socket being
-		// active) which confuses the boot order (the unit
-		// exists before we are fully seeded).
-		isActive, err := sysd.IsActive(unit)
-		if err != nil {
-			return err
-		}
-		serviceUnits := []string{unit}
-		if isActive {
-			// we can never restart the snapd.socket because
-			// this will also bring down snapd itself
-			if unit != "snapd.socket" {
-				if err := sysd.Restart(serviceUnits, 5*time.Second); err != nil {
-					return err
-				}
+	if !opts.Preseeding {
```

Comment:
With emulation mode now handling stop/start/restart can we skip the if check here?